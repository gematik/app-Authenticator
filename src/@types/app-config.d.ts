/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 *
 * ******
 *
 * For additional notes and disclaimer from gematik and in case of changes by gematik find details in the "Readme" file.
 */

declare interface AppConfig {
  /**
   * Whether the window can be manually resized by the user.
   */
  resizable: boolean;
  /**
   * Allow a https page to run JavaScript, CSS or plugins from http URLs.
   */
  allowRunningInsecureContent: boolean;
  /**
   * In production always true, we use this to bypass CORS on dev mode
   */
  webSecurity: boolean;
  /**
   * When 'true', it will install the dev tools
   */
  installDevTools: boolean;
  /**
   * Open development tools on App start
   */
  openDevTools: boolean;
  /**
   * Default window title. Default is `"Electron"`. If the HTML tag `<title>` is
   * defined in the HTML file loaded by `loadURL()`, this property will be ignored.
   */
  title: string;
  /**
   * The window icon. On Windows it is recommended to use `ICO` icons to get the best
   * visual effects, you can also leave it undefined so the executable's icon will be
   * used.
   */
  icon: string;
  /**
   * Window's width in pixels. Default is `800`.
   */
  width: number;
  /**
   * Window's height in pixels. Default is `600`.
   */
  height: number;
  /**
   * Window's minimum width. Default is `0`.
   */
  minWidth: number;
  /**
   * Window's minimum height. Default is `0`.
   */
  minHeight: number;
  /**
   * The bundle app id.
   */
  appId: string;
}

declare module 'appConfig' {
  function appConfigFactory(): AppConfig;

  export = appConfigFactory;
}
