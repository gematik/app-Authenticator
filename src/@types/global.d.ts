/*
 * Copyright 2024 gematik GmbH
 *
 * The Authenticator App is licensed under the European Union Public Licence (EUPL); every use of the Authenticator App
 * Sourcecode must be in compliance with the EUPL.
 *
 * You will find more details about the EUPL here: https://joinup.ec.europa.eu/collection/eupl
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the EUPL is distributed on an "AS
 * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the EUPL for the specific
 * language governing permissions and limitations under the License.ee the Licence for the specific language governing
 * permissions and limitations under the Licence.
 */

import fs from 'fs';
import url from 'url';
import https from 'https';
import os from 'os';
import util from 'util';
import path from 'path';
import { Options } from 'got';
import { TClientRes } from '@/main/services/http-client';
import { P12_VALIDITY_TYPE } from '@/constants';

export declare global {
  interface Window {
    api: {
      on: (eventName: string, callback) => void;
      send: (eventName: string, ...arguments: unknown[]) => unknown;
      sendSync: (eventName: string, ...arguments: unknown[]) => unknown;
      openExternal: (eventName: string) => void;
      focusToApp: () => void;
      isFile: (string: string) => boolean;
      readFileSync: typeof fs.readFileSync;
      copyFileSync: typeof fs.copyFileSync;
      readdirSync: typeof fs.readdirSync;
      existsSync: typeof fs.existsSync;
      unlinkSync: typeof fs.unlinkSync;
      mkdirSync: typeof fs.mkdirSync;
      writeFileSync: typeof fs.writeFileSync;
      pathJoin: typeof path.join;
      pathDirname: typeof path.dirname;
      pathSep: () => string;
      getTmpDir: typeof os.tmpdir;
      utilFormat: typeof util.format;
      nativeURL: typeof url.URL;
      httpsAgent: typeof https.Agent;
      setAppConfigInPreload: (data: Record<string, unknown>) => void;
      setCaChainIdpInPreload: (data: string[]) => void;
      httpGet: (url: string, options?: Options) => Promise<TClientRes>;
      httpPost: (url: string, envelope: any, options?: Options) => Promise<TClientRes>;
      getProcessCwd: () => string;
      getProcessEnvs: () => Record<string, string>;
      createLogZipFile: () => Promise<boolean>;
      isP12Valid: (p12Path: string, password: string) => P12_VALIDITY_TYPE;
      extractValidCertificate: (p12Path: string, password: string) => string;
      isMacOS: () => boolean;
      homedir: () => string;
      readLicenceFile: () => string;
    };
  }
}
