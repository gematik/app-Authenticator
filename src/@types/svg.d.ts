/*
 * Copyright 2024 gematik GmbH
 *
 * The Authenticator App is licensed under the European Union Public Licence (EUPL); every use of the Authenticator App
 * Sourcecode must be in compliance with the EUPL.
 *
 * You will find more details about the EUPL here: https://joinup.ec.europa.eu/collection/eupl
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the EUPL is distributed on an "AS
 * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the EUPL for the specific
 * language governing permissions and limitations under the License.ee the Licence for the specific language governing
 * permissions and limitations under the Licence.
 */

declare module '*.svg' {
  // we use string here instead of XMLDocument to be able to simple replace strings
  const doc: string;
  export default doc;
}
