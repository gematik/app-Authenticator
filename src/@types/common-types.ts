/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 */

import { SweetAlertOptions } from 'sweetalert2';
import { IdpError } from '@/renderer/modules/gem-idp/type-definitions';

export interface TOidcProtocol2UrlSpec {
  challenge_path?: string;
  redirect_automatically?: boolean;
}

export type TUserWarnObject = {
  swalOptions: Partial<SweetAlertOptions>;
  data: Record<string, unknown>;
};

export type TAuthFlowEndState = { isSuccess: boolean; url: string; idpError?: IdpError };

/**
 * Options types for FormInput DropDown element.
 * If you implement anything, please also add types in the FormInput element
 */
export type IFormInputDropDownOptionType = 'standardBool';

/**
 * If option list is not suitable with OptionsType we can give a custom option list for the dropdown
 */
export type IFormInputDropDownOptions = { text: string; value: string | boolean | number };

/**
 * If option list is not suitable with OptionsType we can give a custom option list for the dropdown
 */
export type IFormInputColumnTypes =
  | 'input'
  | 'number'
  | 'password'
  | 'email'
  | 'text'
  | 'drop-down'
  | 'file'
  | 'file-path';

export interface IConfig {
  label: string;
  key: string;
  type: IFormInputColumnTypes;
  required?: boolean;
  iterable?: boolean;
  placeholder?: string;
  optionsType?: IFormInputDropDownOptionType;
  options?: IFormInputDropDownOptions[];
  hide?: boolean;
  validationRegex?: RegExp;
  onChange?: (...args: any[]) => void;
  validateInput?: (value: string) => boolean;
  sanitizeInput?: (value: string) => void;
  infoText?: string;
}

export interface IConfigSection {
  title: string;
  hide?: boolean;
  icon?: string;
  columns: IConfig[];
}

export enum TLS_AUTH_TYPE {
  BasicAuth = 'BasicAuth',
  ServerCertAuth = 'ServerCertAuth',
  ServerClientCertAuth = 'ServerClientCertAuth',
  ServerClientCertAuth_Pfx = 'ServerClientCertAuth_Pfx',
}

export enum KONNEKTOR_VENDORS {
  Rise = 'Rise',
  Koco = 'KoCo',
  Secunet = 'Secunet',
}
