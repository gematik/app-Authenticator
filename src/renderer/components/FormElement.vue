<!--
  - Copyright 2023 gematik GmbH
  -
  - The Authenticator App is licensed under the European Union Public Licence (EUPL); every use of the Authenticator App
  - Sourcecode must be in compliance with the EUPL.
  -
  - You will find more details about the EUPL here: https://joinup.ec.europa.eu/collection/eupl
  -
  - Unless required by applicable law or agreed to in writing, software distributed under the EUPL is distributed on an "AS
  - IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the EUPL for the specific
  - language governing permissions and limitations under the License.ee the Licence for the specific language governing
  - permissions and limitations under the Licence.
  -->
<template>
  <div
    v-if="hide !== true"
    class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
  >
    <div class="text-sm" :for="`input-${name}`">
      <label :id="`form-${name}-label`" class="text-sm"> {{ label }} </label>
      <Tooltip :tooltip-content="infoText"></Tooltip>
    </div>
    <div>
      <div v-if="type === 'input' || type === 'password' || type === 'email' || type === 'number'">
        <input
          :id="`form-${name}`"
          v-model="model[name]"
          :placeholder="placeholder"
          :disabled="disabled"
          :type="type"
          :required="required"
          class="settings-input"
          @change="onLocalChange"
          @blur="onLocalChange"
        />
      </div>
      <div v-if="type === 'text'">
        <textarea
          :id="`form-${name}`"
          v-model="model[name]"
          :placeholder="placeholder"
          :disabled="disabled"
          :required="required"
          class="settings-input"
          @change="onLocalChange"
          @blur="onLocalChange"
        />
      </div>
      <div v-else-if="type === 'drop-down'">
        <select
          :id="`form-${name}`"
          v-model="model[name]"
          :disabled="disabled"
          :required="required"
          class="settings-input"
          @change="onLocalChange"
          @blur="onLocalChange"
        >
          <option v-for="(boolOpt, index) in getOptions()" :key="index" :value="boolOpt.value">
            {{ boolOpt.text }}
          </option>
        </select>
      </div>
      <div v-else-if="type === 'file' || type === 'file-path'">
        <label class="flex row justify-between settings-input w-[233px] leading-[100%] truncate overflow-hidden">
          <span
            v-for="file in iterable ? model[name] : [model[name]]"
            :key="file"
            class="w-[273px] overflow-hidden truncate rtl"
            :title="type === 'file-path' ? file : $t('file_selected')"
          >
            {{ type === 'file-path' ? file : $t('file_selected') }}
          </span>
          <img
            v-if="!model[name] || iterable"
            src="@/assets/file-upload.svg"
            alt="File-Upload Icon"
            class="object-none"
          />
          <input
            v-if="!model[name] || iterable"
            :id="`form-${name}`"
            :disabled="disabled"
            :required="required && !iterable"
            accept=".pem,.p12,.cer"
            title=""
            type="file"
            @change="onFileChange"
          />
          <button v-if="model[name]" type="button" @click="clearValue">
            <svg
              class="h-6 w-6"
              xmlns="http://www.w3.org/2000/svg"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
              aria-hidden="true"
            >
              <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M6 18L18 6M6 6l12 12" />
            </svg>
          </button>
        </label>
      </div>
      <div v-if="!isValid" class="text-error text-[12px] px-[15px] pt-[5px]">{{ $t('value_is_not_valid') }}</div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';
import { IFormInputColumnTypes, IFormInputDropDownOptions, IFormInputDropDownOptionType } from '@/@types/common-types';
import Tooltip from '@/renderer/components/Tooltip.vue';

export default defineComponent({
  name: 'FormElement',
  components: { Tooltip },
  props: {
    model: {
      type: Object,
      required: true,
    },
    required: {
      type: Boolean,
      required: true,
    },
    iterable: {
      type: Boolean,
      required: false,
      default: false,
    },
    type: {
      type: String as PropType<IFormInputColumnTypes>,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    label: {
      type: String,
      required: true,
    },
    optionsType: {
      type: String as PropType<IFormInputDropDownOptionType>,
      required: false,
      default: () => '',
    },
    options: {
      type: Array as PropType<IFormInputDropDownOptions[]>,
      required: false,
      default: () => [],
    },
    placeholder: {
      type: String,
      required: false,
      default: '',
    },
    validationRegex: {
      type: RegExp,
      required: false,
      default: null,
    },
    hide: {
      type: Boolean,
      required: false,
    },
    disabled: {
      type: Boolean,
      required: false,
      default: false,
    },
    infoText: {
      type: String,
      required: false,
      default: '',
    },
    /**
     * Get fired for whole form any time an element changes
     */
    onFormChange: {
      type: Function,
      required: false,
      default: (): void => {},
    },
    /**
     * Get fired only for the element's listener
     */
    onElementChange: {
      type: Function,
      required: false,
      default: (): void => {},
    },
    accept: {
      type: String,
      required: false,
      default: '.pem',
    },
  },
  data() {
    return {
      isValid: true,
    };
  },
  created() {
    this.validate();
  },
  methods: {
    clearValue(): void {
      this.model[this.name] = '';
    },
    /**
     * after each change we call this method and this method fires prop onFormChange
     */
    onLocalChange(): void {
      this.validate();

      this.onFormChange();
    },
    async onFileChange(e: Event): Promise<void> {
      const input = e.target as HTMLInputElement;
      const file = input.files && input.files[0];

      // for file-path type we only need absolute path of the file, not the content
      if (this.type === 'file-path' && file) {
        if (!this.iterable) {
          this.model[this.name] = file?.path;

          this.onElementChange(e);
          this.onFormChange();
          return;
        }

        if (!this.model[this.name].length) {
          this.model[this.name] = [];
        }
        this.model[this.name].push(file?.path);

        this.onElementChange(e);
        this.onFormChange();
        return;
      }

      const reader = new FileReader();
      if (file) {
        reader.readAsText(file);

        reader.onload = () => {
          this.model[this.name] = reader.result;
        };
      }

      this.onElementChange();
      this.onFormChange();
    },

    getOptions(): IFormInputDropDownOptions[] {
      if (this.options?.length) {
        return this.options;
      } else if (this.optionsType) {
        const optionTypes = {
          standardBool: [
            { text: this.$t('enabled'), value: true },
            { text: this.$t('disabled'), value: false },
          ],
        };

        return optionTypes[this.optionsType];
      }
      return [];
    },
    validate() {
      if (this.model[this.name] && this.validationRegex && !this.validationRegex.test(this.model[this.name])) {
        this.isValid = false;
      } else {
        this.isValid = true;
      }
    },
  },
});
</script>

<style lang="scss" scoped>
.form-element-container:last-of-type {
  @apply rounded-[8px];
}

.inner-box-shadow {
  box-shadow: inset -1px 0 0 #e4e7ec, inset 1px 0 0 #e4e7ec, inset 0 -1px 0 #e4e7ec;
}

.settings-input {
  @apply h-[36px] rounded-[8px] px-[12px] py-[8px] bg-neutral border-inputBorder border;
  @apply w-[333px] leading-[100%];
  /* focus: focus: */
}

.settings-input:focus {
  @apply bg-inputBgFocus border-inputBorderFocus outline-none;
}

.rtl {
  direction: rtl;
}

input[type='file'] {
  opacity: 0;
  position: absolute;
}
</style>
