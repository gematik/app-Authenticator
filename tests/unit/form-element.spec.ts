/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 *
 * ******
 *
 * For additional notes and disclaimer from gematik and in case of changes by gematik find details in the "Readme" file.
 */

import { mount } from '@vue/test-utils';
import FormElement from '@/renderer/components/FormElement.vue';
import i18n from '@/renderer/i18n';
import { IFormInputColumnTypes, IFormInputDropDownOptions, IFormInputDropDownOptionType } from '@/@types/common-types';

type IFormElementProps = {
  model: Record<string, unknown>;
  required: boolean;
  name: string;
  label: string;
  placeholder: string;
  type: IFormInputColumnTypes;
  options?: IFormInputDropDownOptions[];
  optionsType?: IFormInputDropDownOptionType;
};

describe('form element', () => {
  const props: IFormElementProps = {
    model: {},
    required: true,
    name: 'name',
    label: 'Awesome Label',
    type: 'text',
    placeholder: 'Awesome Value',
  };

  const renderFormElementForProps = (props: IFormElementProps) => {
    return mount(FormElement, { props, global: { plugins: [i18n] } });
  };

  it('renders input field', async function () {
    props.type = 'input';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders text field', async function () {
    props.type = 'text';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders number field', async function () {
    props.type = 'number';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders drop-down for standard options', async function () {
    props.type = 'drop-down';
    props.optionsType = 'standardBool';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders drop-down for custom options', async function () {
    props.type = 'drop-down';
    props.options = [{ text: 'Awesome', value: 'awesome' }];
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders email field', async function () {
    props.type = 'email';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders file field', async function () {
    props.type = 'file';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders file-path field', async function () {
    props.type = 'file-path';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });

  it('renders file-path field', async function () {
    props.type = 'password';
    expect(renderFormElementForProps(props).element).toMatchSnapshot();
  });
});
