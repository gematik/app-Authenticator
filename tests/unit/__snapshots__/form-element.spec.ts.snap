// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form element renders drop-down for custom options 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <!--v-if-->
    <div>
      <select
        class="settings-input"
        id="form-name"
        required=""
      >
        
        <option
          value="awesome"
        >
          Awesome
        </option>
        
      </select>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders drop-down for standard options 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <!--v-if-->
    <div>
      <select
        class="settings-input"
        id="form-name"
        required=""
      >
        
        <option
          value="true"
        >
          Aktiviert
        </option>
        <option
          value="false"
        >
          Deaktiviert
        </option>
        
      </select>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders email field 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <div>
      <input
        class="settings-input"
        id="form-name"
        placeholder="Awesome Value"
        required=""
        type="email"
      />
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders file field 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <!--v-if-->
    <div>
      <label
        class="flex row justify-between settings-input w-[233px] leading-[100%] truncate overflow-hidden"
      >
        
        <span
          class="w-[273px] overflow-hidden truncate rtl"
          title="Datei Ausgewählt"
        >
          Datei Ausgewählt
        </span>
        
        <img
          alt="File-Upload Icon"
          class="object-none"
        />
        <input
          accept=".pem,.p12,.cer"
          id="form-name"
          required=""
          title=""
          type="file"
        />
        <!--v-if-->
      </label>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders file-path field 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <!--v-if-->
    <div>
      <label
        class="flex row justify-between settings-input w-[233px] leading-[100%] truncate overflow-hidden"
      >
        
        <span
          class="w-[273px] overflow-hidden truncate rtl"
        />
        
        <img
          alt="File-Upload Icon"
          class="object-none"
        />
        <input
          accept=".pem,.p12,.cer"
          id="form-name"
          required=""
          title=""
          type="file"
        />
        <!--v-if-->
      </label>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders file-path field 2`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <div>
      <input
        class="settings-input"
        id="form-name"
        placeholder="Awesome Value"
        required=""
        type="password"
      />
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders input field 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <div>
      <input
        class="settings-input"
        id="form-name"
        placeholder="Awesome Value"
        required=""
        type="input"
      />
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders number field 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <div>
      <input
        class="settings-input"
        id="form-name"
        placeholder="Awesome Value"
        required=""
        type="number"
      />
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`form element renders text field 1`] = `
<div
  class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
>
  <div
    class="text-sm"
    for="input-name"
  >
    <label
      class="text-sm"
      id="form-name-label"
    >
      Awesome Label
    </label>
    <div
      class="tooltip"
    >
      <img
        alt=""
      />
      <span
        class="tooltiptext"
      />
    </div>
  </div>
  <div>
    <!--v-if-->
    <div>
      <textarea
        class="settings-input"
        id="form-name"
        placeholder="Awesome Value"
        required=""
      />
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;
